### ISSUE TITLE
Hot reloading working inconsistently

### ISSUE THREAD

**Issue by briangwaltney** [vitest-dev/vitest#2640]
### Describe the bug

Hot reload failure example
This minimal reproduction shows vitest not hot reloading as expected.

Sporadically, renaming functions, changing implementations, and just overall adjusting the code does not cause the tests to rerun properly.

Sometimes it works, sometimes it doesn't. As soon as it stops working, you can go to the terminal running vitest, press a to rerun all tests and you will continue to get the error.

Upon returning to the code, you can save the file, and the tests will rerun properly.

Screen recording included in repo as failureExample.mov.

I have Wallaby installed and that is shown in the video, but Vitest functions the same with or without Wallaby.

### Reproduction

https://github.com/briangwaltney/slow-vitest-example

### System Info

```shell
System:
    OS: macOS 13.0.1
    CPU: (10) x64 Apple M1 Pro
    Memory: 15.68 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.16.0 - ~/.nvm/versions/node/v16.16.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.16.0/bin/yarn
    npm: 9.1.3 - ~/.nvm/versions/node/v16.16.0/bin/npm
  Browsers:
    Brave Browser: 108.1.46.153
    Chrome: 108.0.5359.124
    Firefox: 107.0.1
    Safari: 16.1
  npmPackages:
    vitest: ^0.26.3 => 0.26.3
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
Fixes by https://github.com/vitest-dev/vitest/commit/9967645ab7ade16a2b2495c7ee0277d180f866f1

**Comment by qc-z** [comment]
> Fixed in [9967645](https://github.com/vitest-dev/vitest/commit/9967645ab7ade16a2b2495c7ee0277d180f866f1)

occasionally no work in 0.27.1
